library/src/updater_lock.rs

```diff
@@ -0,0 +1,54 @@
+use crate::updater::UpdateError;
+
+// This file's job is to handle the boiler plate around locking for the
```

Comment:
I'd add a note about why we need to lock, e.g. "only one thread can mutate the file system at a time"